core/domain/exp_domain.py

```diff
@@ -3519,3 +3519,226 @@ def to_metadata_dict(self):
             'title': self.title,
             'objective': self.objective,
         }
+
+
+class StateIdMapping(object):
+    """Domain object for state id mapping model."""
+
+    def __init__(
+            self, exploration_id, exploration_version, state_name_to_ids,
+            latest_state_id_used):
+        """Initialize state id mapping model object.
+
+        Args:
+            exploration_id: str. The exploration id whose state names are
+            mapped.
+            exploration_version: int. The version of the exploration.
+            state_name_to_ids: dict. A dictionary mapping a state name to
+                unique ID.
+            latest_state_id_used: int. Last used state id.
+        """
+        self.exploration_id = exploration_id
+        self.exploration_version = exploration_version
+        self.state_name_to_ids = state_name_to_ids
+        self.latest_state_id_used = latest_state_id_used
+
+    @staticmethod
+    def has_state_changed_significantly(old_state, new_state):
+        """Checks whether there is any significant change in old and new version
+        of the given state.
+
+        A siginificant change has happend when interaction id has been changed
+        in new version of the state.
+
+        Args:
+            old_state: dict. Old version of the state.
+            new_state: dict. New version of the state.
+
+        Returns:
+            bool. A boolean indicating whether change is significant or not.
+        """
+        if old_state.interaction.id != new_state.interaction.id:
+            # State has changed significantly is interaction id in new state is
+            # different from old state.
+            return True
+
+    @classmethod
+    def get_state_id_mapping(
+            cls, old_exploration, new_exploration, change_list):
+        """Get state name to id mapping for new exploration's states.
+
+        Args:
+            old_exploration: Exploration. Old version of the exploration.
+            new_exploration: Exploration. New version of the exploration.
+            change_list: list(dict). List of changes between old and new
+                exploration.
+
+        Returns:
+            StateIdMapping. Domain object for state id mapping.
+        """
+        new_state_names = []
+        state_id_to_names = {}
+
+        # Get state id mapping model for old exploration.
+        model = cls.get_state_id_mapping_from_model(
+            exp_models.StateIdMappingModel.get_state_id_mapping_model(
+                old_exploration.id, old_exploration.version, strict=True))
+
+        # Generate state id to name mapping dict for old exploration.
+        for state_name in old_exploration.states:
+            state_id_to_names[model.get_state_id(state_name)] = state_name
```

Comment:
Do stuff like get_state_id on the domain object, not the model. The only thing the model should be used for is to initialize the domain object.

Reply:
Done. Actually, it was domain object but the name was model.